### ISSUE TITLE
Reproducibility issue

### ISSUE THREAD

**Issue by fabianrost84** [rfechtner/pypairs#21]
If I calculate cell cycle scores twice, I do not get the same results. Minimal example:
```python
from pypairs import pairs, datasets

# Load samples from the oscope scRNA-Seq dataset with known cell cycle
training_data = datasets.leng15(mode='sorted')

# Run sandbag() to identify marker pairs
marker_pairs = pairs.sandbag(training_data, fraction=0.6)

# Load samples from the oscope scRNA-Seq dataset without known cell cycle
testing_data = datasets.leng15(mode='unsorted')

# Run cyclone() score and predict cell cycle classes
result = pairs.cyclone(testing_data, marker_pairs)
result2 = pairs.cyclone(testing_data, marker_pairs)

# Further downstream analysis
print(result.head())
print(result2.head())
```
Output
```
               G2M      S   G1 max_class cc_prediction
H1_Exp1.001  0.028  0.134  0.0         S             S
H1_Exp1.002  0.016  0.969  0.0         S             S
H1_Exp1.003  1.000  0.000  0.0       G2M           G2M
H1_Exp1.004  0.000  1.000  0.0         S             S
H1_Exp1.006  0.000  0.996  0.0         S             S
               G2M      S     G1 max_class cc_prediction
H1_Exp1.001  0.035  0.147  0.000         S             S
H1_Exp1.002  0.004  0.978  0.000         S             S
H1_Exp1.003  1.000  0.000  0.000       G2M           G2M
H1_Exp1.004  0.001  1.000  0.000         S             S
H1_Exp1.006  0.000  0.995  0.001         S             S
```
Notice that the scores in the two table are not the same.
I'm working with pypairs version v3.2 installed via pip.

**Comment by rfechtner** [comment]
@fabianrost84 This is expected as the score is calculated by shuffling gene pairs and comparing the actual number matched marker pairs against the shuffled distribution. You can reduce the variance by increasing the _iterations_ parameter when using _cyclone()_. This makes the distribution more robust but takes longer time to compute.

**Comment by fabianrost84** [comment]
Thanks for the explanation. I can see the shuffling is done here:
https://github.com/rfechtner/pypairs/blob/8bc94e3eb3590df6e25fee34f1a2ecf501e7b471/pypairs/tools/cyclone.py#L227
However, I think, it would be great if pypairs would set a random seed such that the result is reproducible. Would you agree? Then, I might open a PR.

**Comment by rfechtner** [comment]
Seems like a good idea! Makes it easier to share results.

**Comment by fabianrost84** [comment]
Resolved by #22. Thanks @rfechtner!
